Jenkinsfile

```diff
@@ -229,6 +229,161 @@ def runCypressE2e(){
         junit '**/e2e/**/*.xml'
     }
 }
+
+void postFailureFiexcodeScanAndReport(){
+    def email=''
+    if(env.CHANGE_AUTHOR){
+        def author=env.CHANGE_AUTHOR.toString().trim().toLowerCase()
+        email=getEmailFromGITUser author
+    }
+    else {
+       email=Email
+    }
+    sendMail email,"<h3>Pipeline FAILED in stage ${STAGE_NAME} <a href=${BUILD_URL}/console> Check Console Output Here</a></h4>",false," pipeline Failed in ${STAGE_NAME} "
+}
+
+void fiexcodeScanAndReport(){
+    def palamida_url="http://palamida-2.marklogic.com:8888"
+    def inventoriesRprt=''
+
+    withCredentials([string(credentialsId: 'palamida_jwt', variable: 'jwt')]) {
+
+        StartScan(baseUrl: palamida_url, projectName: 'ml-dhf', token: jwt)
+
+        inventoriesRprt = sh(returnStdout: true, script: '''curl -X GET ''' + palamida_url + '''/codeinsight/api/project/inventory/21 -H "Authorization: Bearer $jwt" ''')
+    }
+
+    writeFile(file: 'inventoriesRprt.report.json', text: inventoriesRprt)
+
+    def slurper = new JsonSlurperClassic().parseText(inventoriesRprt.toString().trim())
+
+    def failures = []
+    slurper.inventoryItems.each { key, value ->
+        key.vulnerabilities.each { key1, value1 ->
+            if(key1.vulnerabilityCvssV3Severity == 'CRITICAL' || key1.vulnerabilityCvssV3Severity == 'HIGH'){
+                def msg = " inventory name: $key.name  volnarability name: $key1.vulnerabilityName with Severity V2: $key1.vulnerabilityCvssV2Severity with Severity V3: $key1.vulnerabilityCvssV3Severity "
+                failures.add(msg)
+
+            }
+        }}
+
+    archiveArtifacts artifacts: 'inventoriesRprt.report.json'
+
+    //Send email
+    if(failures.size()!=0){
+        def body=''
+        failures.each { msg -> body=body+msg + '\n' }
+        sendMail 'Kavitha.Sivagnanam@marklogic.com,yakov.feldman@marklogic.com', body,false,'Flexcode FAILED. HIGH vulnerabilities','inventoriesRprt.report.json'
```

Comment:
you don't need to send the email twice. An email will be sent in the post stage. If required include Kavitha to the recipient list.

Reply:
Then email in `post` would need to include attached report and P1 vulnerabilities info in the body of the email. Is it OK?

Reply:
yes